docs/guided-remediation.md

```diff
@@ -0,0 +1,247 @@
+
+### Vulnerability selection
+
+The following flags may be used to filter which vulnerabilities will be selected for remediation:
+
+- `--max-depth=<value>`: The maximum dependency depth to consider vulnerabilities from.
+
+  For example, `--max-depth=1` will only consider vulnerabilities affecting direct dependencies, while `--max-depth=2` will consider up to the direct dependencies of your direct dependencies. Setting `--max-depth=-1` will consider all dependencies, regardless of depth.
+
+- `--min-severity=<value>`: The minimum CVSS score of vulnerabilities to consider.
+
+  For example, `--min-severity=7.5` will consider only vulnerabilities with CVSS a score of 7.5 and above. If the OSV record for the vulnerability does not have a CVSS score, it will **not** be excluded.
+
+- `--ignore-dev`: Whether to exclude vulnerabilities from dependencies that are only used in `devDependencies`.
+- `--ignore-vulns=<comma-separated list of IDs>`: A list of OSV IDs to exclude from consideration.
+- `--vulns=<comma-separated list of IDs>`: An explicit list of OSV IDs that should exclusively be considered, as long as the other criteria are satisfied.
+
+A vulnerability is only considered if it satisfies all the conditions set by these flags.
+
+### Dependency upgrade options
+
+The following flags may be used to limit the patches allowed for your dependencies:
+
+- `--disallow-major-upgrades`: Do no allow patches that would result in the major version number of any dependency from being changed.
+- `--disallow-package-upgrades=<comma-separated list of package names>`: Do no allow patches to any of the listed packages.
+
+### Data source
+
+By default, we use the [deps.dev API](https://docs.deps.dev/api/v3alpha/) to find version and dependency information of packages during remediation.
+
+If instead you'd like to use your ecosystem's native registry API (e.g. `https://registry.npmjs.org`), you can use the `--data-source=native` flag. `osv-scanner fix` will attempt to use the authentication specified by the native tooling (e.g. `npm config`)
+
+{: .highlight }
+If your project uses mirrored or private registries, you will need to use `--data-source=native`
+
+{: .note }
+
+> The subcommand caches the requests it makes in `package.json.resolve.deps` (deps.dev) and `package.json.resolve.npm` (native npm).
+>
+> The native npm cache will store the addresses of private registries used, though not any authentication information.
+
+## Known issues
+
+- The subcommand does not use the `osv-scanner.toml` configuration. Use the `--ignore-vulns` flag instead.
+
+### npm
+
+- Non-registry dependencies (local paths, URLs, Git, etc.) are not evaluated.
+- `peerDependencies` are not properly considered during dependency resolution (treated as if using `--legacy-peer-deps`).
+- `overrides` are ignored during dependency resolution.
+- The `node_modules/` in workspaces are not deleted when relocking, which may impact the resulting dependency graph.
```

Comment:
Also, should we add a section specifically to call out workspaces and how we handle that?

Reply:
Expanded upon the workspaces in its own sub-sub-sub-heading :)